libs/RobotKit/include/RobotController.h

```diff
@@ -0,0 +1,50 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _LEKA_OS_ROBOT_CONTROLLER_H_
+#define _LEKA_OS_ROBOT_CONTROLLER_H_
+
+#include "StateMachine.h"
+#include "interface/RobotController.h"
+#include "interface/drivers/Battery.h"
+
+namespace leka {
+
+template <typename sm_t = boost::sml::sm<system::robot::StateMachine>>
+class RobotController : public interface::RobotController
+{
+  public:
+	sm_t state_machine {static_cast<interface::RobotController &>(*this)};
+
+	explicit RobotController(interface::Battery &battery) : _battery(battery) {};
+
+	auto isCharging() -> bool final { return _battery.isCharging(); };
+
+	template <typename eventType>
+	void raise()
+	{
+		state_machine.process_event(eventType {});
+	};
```

Comment:
un neat trick pour les fonctions où tu n'as pas vraiment besoin du type (comme là), c'est d'utiliser `auto` pour les arguments:

```suggestion
void raise(auto event) { state_machine.process_event(event); };
```

ce qui permet de faire... (voir comment suivant)